File: cloud-collect-core/src/main/java/com/windea/demo/cloudcollect/core/repository/CommentRepository.java
package com.windea.demo.cloudcollect.core.repository;

import com.windea.demo.cloudcollect.core.domain.entity.Comment;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;

public interface CommentRepository extends JpaRepository<Comment, Long> {
	Page<Comment> findByCollect_Id(Long collectId, Pageable pageable);

	long countByCollect_Id(Long collectId);

	Page<Comment> findBySponsorByUser_Id(Long sponsorByUserId, Pageable pageable);

	long countBySponsorByUser_Id(Long sponsorByUserId);

	Page<Comment> findByReplyToComment_Id(Long replyToCommentId, Pageable pageable);

	long countByReplyToComment_Id(Long replyToCommentId);
}

File: cloud-collect-core/src/main/kotlin/com/windea/demo/cloudcollect/core/repository/CommentRepository.kt
package com.windea.demo.cloudcollect.core.repository

import com.windea.demo.cloudcollect.core.domain.entity.*
import org.springframework.data.domain.*
import org.springframework.data.jpa.repository.*

interface CommentRepository : JpaRepository<Comment, Long> {
	//需要按id倒序排序显示
	fun findAllByCollectIdOrderByIdDesc(collectId: Long, pageable: Pageable): Page<Comment>
	
	//需要按id倒序排序显示
	fun findAllBySponsorByUserIdOrderByIdDesc(sponsorByUserId: Long, pageable: Pageable): Page<Comment>
	
	//需要按id倒序排序显示
	fun findAllByReplyToCommentIdOrderByIdDesc(replyToCommentId: Long, pageable: Pageable): Page<Comment>
	
	fun countByCollectId(collectId: Long): Long
	
	fun countBySponsorByUserId(sponsorByUserId: Long): Long
	
	fun countByReplyToCommentId(replyToCommentId: Long): Long
}
